### PR TITLE
verify upgrade timeout path

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

- adds functionality to verify the timeout set by a chain initiating a channel upgrade

closes: #1610 

### Commit Message / Changelog Entry

NA

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by chatton — APPROVED** [review]
LGTM! thanks for taking this one 👍

**Review by damiannolan — APPROVED** [review]
Awesome, nice work!

File: modules/core/03-connection/keeper/verify_test.go [hunk 1]
```diff
@@ -682,6 +682,102 @@ func (suite *KeeperTestSuite) TestVerifyNextSequenceRecv() {
...
+
+			upgradeTimeoutKey := host.ChannelUpgradeTimeoutKey(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
+			proof, proofHeight := suite.chainA.QueryProof(upgradeTimeoutKey)
+
+			// The VerifyChannelUpgrade step is called by chain B, and will prove the timeout values set by chain A
+			// have not passed on chain B.
+			err = suite.chainB.GetSimApp().IBCKeeper.ConnectionKeeper.VerifyChannelUpgradeTimeout(
+				suite.chainB.GetContext(),
+				path.EndpointB.GetConnection(),
+				malleateHeight(proofHeight, 0),
```
Comment (crodriguezvega) [inline modules/core/03-connection/keeper/verify_test.go:764]: Just a nit: is the call to `malleateHeight` necessary is `diff` is 0?

**Review by crodriguezvega — APPROVED** [review]
Just left a nit, otherwise looks good to me!

**Pull request merged**